libraries/ms-common/inc/spi.h

```diff
@@ -0,0 +1,111 @@
+// SPI HAL Interface
```

Comment:
It's really important that you have an include guard or use `#pragma once`. Since `#include` is basically just a copy + paste, including a header multiple times without an include guard would result in duplicate function prototypes, defines, and all the other things that are included in a header.

`#pragma once` is a nonstandard preprocessor directive that does the same thing as the standard `#ifndef` include guard.